internal/pkg/vcsutil/vcs_util.go

```diff
@@ -0,0 +1,276 @@
+package vcsutil
+
+import (
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/hashicorp/go-hclog"
+	"github.com/pkg/errors"
+)
+
+var (
+	sshRemoteRegexp  *regexp.Regexp
+	httpRemoteRegexp *regexp.Regexp
+)
+
+func init() {
+	// Regex matching http/https remotes, tokenizing the unique components for replacement.
+	sshRemoteRegexp, _ = regexp.Compile(`git@(.*?):(.*\.git)`)                 // Example: git@git.test:testorg/testrepo.git
+	httpRemoteRegexp, _ = regexp.Compile(`http[s]?:\/\/(.*?\..*?)\/(.*\.git)`) // Example: https://git.test/testorg/testrepo.git
+}
+
+type VcsChecker struct {
```

Comment:
Given this is a Git-specific, I'd just name this `Git` or `Repo` (I prefer the latter and just making this whole package Git-specific. I've done "VCS abstraction" in the past and its generally not worth it)